embulk-input-s3/src/test/java/org/embulk/input/s3/TestRetrySupportPluginTask.java

```diff
@@ -0,0 +1,80 @@
+package org.embulk.input.s3;
```

Comment:
I think it's not redundant to have more tests, an also same kind as TestHttpProxy or TestFileList.

Reply:
TestFileList tests FileList, not FileList.Task, it has additional logic.
TestHttpProxy, this one has historical reason, HttpProxy started not as a `org.embulk.config.Task` but a plain object with some parsing implementation (https://github.com/embulk/embulk-input-s3/commit/5d5b883ca74335d35043de54bda66f180e7f45a7), its unit test was modified to adapt but not remove, it is understandable, but we shall not introduce same kind of tests for just because of this.
Test redundancy is actually a good thing, but it is for when we cover our own code path, not the library implementation. This test is equal for testing putting key-value on HashMap and assert for the same value when looking up for that key.

Reply:
@sakama your opinion?

Reply:
Seems unnecessary for now.
I think such kind of tests are necessary when something config expect to have specific value at another config.
For example, following code will throw ConfigException when dependent config has invalid value.
We'd like to check such cases to prevent unintended changes.

However, current test doesn't have such dependency check. So I think you can eliminate this test.

* Implementation
```java
@Config("auth_method")
@ConfigDefault("\"private_key\"")
String getAuthMethod();

@Config("json_keyfile")
@ConfigDefault("null")
Optional<LocalFile> getJsonKeyfile();

@Override
public ConfigDiff transaction(ConfigSource config,
                              FileInputPlugin.Control control)
{
    ...
    // Validate auth method
    if (task.getAuthMethod.equals("private_key") && !task.getJsonKeyfile().isPresent()) {
        throw new ConfigException("json_keyfile is necessary when auth_method is private_key");
    }
    ...
}
```

* Test code
```java
@Test(expected = ConfigException.class)
public void testAuthMethod() throws Exception
{
    ConfigSource config = Exec.newConfigSource()
                .set("auth_method", "private_key")
                .set("json_keyfile", "")
}
```